src/main/java/uk/gov/companieshouse/confirmationstatementapi/controller/ConfirmationStatementController.java

```diff
@@ -84,9 +84,14 @@ public ConfirmationStatementController(ConfirmationStatementService confirmation
         logMap.put(TRANSACTION_ID_KEY, transactionId);
         ApiLogger.infoContext(requestId, "Calling service to retrieve confirmation statement data", logMap);
 
-        var serviceResponse = confirmationStatementService.getConfirmationStatement(submissionId);
-        return serviceResponse.<ResponseEntity<Object>>map(ResponseEntity::ok)
-                .orElseGet(() -> ResponseEntity.notFound().build());
+        try {
+            var serviceResponse = confirmationStatementService.getConfirmationStatement(submissionId);
+            return serviceResponse.<ResponseEntity<Object>>map(ResponseEntity::ok)
+                    .orElseGet(() -> ResponseEntity.notFound().build());
+        } catch (SubmissionNotFoundException e) {
```

Comment:
Once the confirmation statement service is just returning an optional.

You can do an 
`if (serviceResponse.isEmpty()) then LogError(); return NotFoundResponse();`

and `else return successResponse();`